<div style="color:white;
           display:fill;
           border-radius:5px;
           background-color:green;
           font-size:200%;
           font-family:Serif;
           letter-spacing:0.5px">

<p style="padding: 10px;
          color:white;
          font-size:120%;
          text-align:left;">
Import libraries
</p>
</div>

In [97]:
import pandas as pd
import numpy as np
import plotly.express as px
import warnings
warnings.filterwarnings('ignore')

<div style="color:white;
           display:fill;
           border-radius:5px;
           background-color:green;
           font-size:200%;
           font-family:Serif;
           letter-spacing:0.5px">

<p style="padding: 10px;
          color:white;
          font-size:120%;
          text-align:left;">
EDA
</p>
</div>

In [98]:
df = pd .read_csv('netflix1.csv')
# The first 5 values
df.head(5)

show_id     type                             title         director  \
0      s1    Movie              Dick Johnson Is Dead  Kirsten Johnson   
1      s3  TV Show                         Ganglands  Julien Leclercq   
2      s6  TV Show                     Midnight Mass    Mike Flanagan   
3     s14    Movie  Confessions of an Invisible Girl    Bruno Garotti   
4      s8    Movie                           Sankofa     Haile Gerima   

         country date_added  release_year rating  duration  \
0  United States  9/25/2021          2020  PG-13    90 min   
1         France  9/24/2021          2021  TV-MA  1 Season   
2  United States  9/24/2021          2021  TV-MA  1 Season   
3         Brazil  9/22/2021          2021  TV-PG    91 min   
4  United States  9/24/2021          1993  TV-MA   125 min   

                                           listed_in  
0                                      Documentaries  
1  Crime TV Shows, International TV Shows, TV Act...  
2                 TV Dramas, TV Horror, TV Mysteries  
3                 Children & Family Movies, Comedies  
4   Dramas, Independent Movies, International Movies

In [99]:
# The last 5 values
df.tail(5)

show_id     type               title   director        country  \
8785   s8797  TV Show          Yunus Emre  Not Given         Turkey   
8786   s8798  TV Show           Zak Storm  Not Given  United States   
8787   s8801  TV Show  Zindagi Gulzar Hai  Not Given       Pakistan   
8788   s8784  TV Show                Yoko  Not Given       Pakistan   
8789   s8786  TV Show                 YOM  Not Given       Pakistan   

      date_added  release_year rating   duration  \
8785   1/17/2017          2016  TV-PG  2 Seasons   
8786   9/13/2018          2016  TV-Y7  3 Seasons   
8787  12/15/2016          2012  TV-PG   1 Season   
8788   6/23/2018          2016   TV-Y   1 Season   
8789    6/7/2018          2016  TV-Y7   1 Season   

                                              listed_in  
8785                  International TV Shows, TV Dramas  
8786                                           Kids' TV  
8787  International TV Shows, Romantic TV Shows, TV ...  
8788                                           Kids' TV  
8789                                           Kids' TV

In [100]:
# Basic info about data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8790 non-null   object
 1   type          8790 non-null   object
 2   title         8790 non-null   object
 3   director      8790 non-null   object
 4   country       8790 non-null   object
 5   date_added    8790 non-null   object
 6   release_year  8790 non-null   int64 
 7   rating        8790 non-null   object
 8   duration      8790 non-null   object
 9   listed_in     8790 non-null   object
dtypes: int64(1), object(9)
memory usage: 686.8+ KB


In [101]:
df['date_added'] = pd.to_datetime(df['date_added'])

In [102]:
df.dtypes

show_id                 object
type                    object
title                   object
director                object
country                 object
date_added      datetime64[ns]
release_year             int64
rating                  object
duration                object
listed_in               object
dtype: object

In [103]:
# Statistical informations. Notice it is only work for int data 
df.describe()

release_year
count   8790.000000
mean    2014.183163
std        8.825466
min     1925.000000
25%     2013.000000
50%     2017.000000
75%     2019.000000
max     2021.000000

In [104]:
# Is there any missing values ?
print('Is there any missing values?',df.isnull().sum().any())

Is there any missing values? False


In [105]:
# df columns
df.columns

Index(['show_id', 'type', 'title', 'director', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in'],
      dtype='object')

In [106]:
# Drop unnecessary columns
df.drop(['show_id'], axis=1, inplace=True)
df.columns

Index(['type', 'title', 'director', 'country', 'date_added', 'release_year',
       'rating', 'duration', 'listed_in'],
      dtype='object')

In [107]:
# Shape of data (Rows , columns)
print("The shape of data is ",df.shape)

The shape of data is  (8790, 9)


In [108]:
df.duplicated().value_counts()

False    8787
True        3
dtype: int64

In [109]:
df['title'].nunique()

8787

Total number of rows are 8790 and 8787 is the no. of the unique titles means we have three such titles which are duplicated or null

In [110]:
df=df.drop_duplicates(subset='title')

In [111]:
df.duplicated().value_counts()

False    8787
dtype: int64

In [112]:
# What is more ? TV show or Moives 
df['type'].value_counts()

Movie      6124
TV Show    2663
Name: type, dtype: int64

In [113]:
# Which country produce more 
df['country'].value_counts()

United States     3240
India             1056
United Kingdom     638
Pakistan           420
Not Given          287
                  ... 
Iran                 1
West Germany         1
Greece               1
Zimbabwe             1
Soviet Union         1
Name: country, Length: 86, dtype: int64

From above: The United States has the largest production

In [114]:
df['release_year'].value_counts()

2018    1144
2017    1030
2019    1029
2020     953
2016     901
        ... 
1966       1
1959       1
1925       1
1947       1
1961       1
Name: release_year, Length: 74, dtype: int64

Most of the productions were in 2018

In [115]:
df['rating'].value_counts()

TV-MA       3205
TV-14       2155
TV-PG        861
R            798
PG-13        490
TV-Y7        333
TV-Y         306
PG           287
TV-G         220
NR            79
G             41
TV-Y7-FV       6
NC-17          3
UR             3
Name: rating, dtype: int64

Most reviews are TV-MA

In [116]:
df['listed_in'].value_counts()

Dramas, International Movies                                362
Documentaries                                               359
Stand-Up Comedy                                             334
Comedies, Dramas, International Movies                      274
Dramas, Independent Movies, International Movies            252
                                                           ... 
Anime Features                                                1
Action & Adventure, Horror Movies, Independent Movies         1
Action & Adventure, Classic Movies, International Movies      1
Cult Movies, Independent Movies, Thrillers                    1
Classic & Cult TV, Crime TV Shows, TV Dramas                  1
Name: listed_in, Length: 513, dtype: int64

The most classifications are Dramas, International Movies

In [117]:
df['duration'].value_counts()

1 Season      1790
2 Seasons      421
3 Seasons      198
90 min         152
97 min         146
              ... 
5 min            1
16 min           1
186 min          1
193 min          1
11 Seasons       1
Name: duration, Length: 220, dtype: int64

Most productions consist of one season

In [118]:
country = df.groupby(['country','type'])['type'].count().sort_values().reset_index(name='count')
country

country     type  count
0          Zimbabwe    Movie      1
1            Cyprus  TV Show      1
2    Czech Republic  TV Show      1
3      Soviet Union    Movie      1
4           Somalia    Movie      1
..              ...      ...    ...
133        Pakistan  TV Show    349
134  United Kingdom    Movie    387
135   United States  TV Show    845
136           India    Movie    975
137   United States    Movie   2395

[138 rows x 3 columns]

Most production in the United States,India and United Kingdom is Movies

In [119]:
cu =  df.groupby(['country','listed_in'])['listed_in'].count().sort_values().reset_index(name='count')
cu

country                                          listed_in  count
0         Argentina  Action & Adventure, Comedies, International Mo...      1
1          Pakistan  Anime Series, International TV Shows, TV Thril...      1
2          Pakistan  Anime Series, International TV Shows, Romantic...      1
3          Pakistan  Anime Series, Crime TV Shows, International TV...      1
4          Pakistan                      Anime Features, Documentaries      1
...             ...                                                ...    ...
2015          India                       Dramas, International Movies    122
2016  United States                 Children & Family Movies, Comedies    122
2017          India             Comedies, Dramas, International Movies    124
2018  United States                                    Stand-Up Comedy    216
2019  United States                                      Documentaries    269

[2020 rows x 3 columns]

Most production in the United States are: Documentaries and Stand-Up Comedy

Most production in the India are: Dramas, International Movies and Comedies

<div style="color:white;
           display:fill;
           border-radius:5px;
           background-color:green;
           font-size:200%;
           font-family:Serif;
           letter-spacing:0.5px">

<p style="padding: 10px;
          color:white;
          font-size:120%;
          text-align:left;">
Visualization
</p>
</div>

In [120]:
fig = px.pie(df , 'type' , title = 'The biggest show type')
fig.show()

In [121]:
px.histogram(data_frame=df,x='country',color='country',title='The amount of productions for countries')

In [122]:
px.histogram(data_frame=df,x='rating',color='rating',title='Clarification of ratings')

From the above figure we noticed that 3205 show is for Mature Audiences and 2157 for may not be suitable for ages under 14.

In [123]:
px.histogram(data_frame=df,x='rating',color='type',title='Show type and it`s rating')

In [124]:
px.histogram(data_frame=df,x='rating',color='country',title="Ratings of the country's productions")

In [125]:
top10_rating = df['listed_in'].value_counts().head(10)
fig = px.bar(top10_rating , x=top10_rating.index , y=top10_rating.values,title='Top 10 categories of shows')
fig.show()

In [126]:
top10_director = df['director'].value_counts().drop("Not Given").head(10)
fig = px.bar(top10_director , x=top10_director.index , y=top10_director.values,title='Most productive directors',width=1000,height=400,color_discrete_sequence=["green"])
fig.show()



In [127]:
top10_years = df['release_year'].value_counts().head(10)
fig = px.bar(top10_years , x=top10_years.index , y=top10_years.values,title='Top 10 years in which productions took place')
fig.show()

In [128]:
top10_duration = df['duration'].value_counts().head(10)
fig = px.bar(top10_duration , x=top10_duration.index , y=top10_duration.values,title='Most 10 durations')
fig.show()

<div style="color:white;
           display:fill;
           border-radius:5px;
           background-color:green;
           font-size:200%;
           font-family:Serif;
           letter-spacing:0.5px">

<p style="padding: 10px;
          color:white;
          font-size:120%;
          text-align:left;">
Analysis report
</p>
</div>

1) Among the countries United states stands on number one position followed by India and UK.These countries are known for producing many more series and movies as compared to other countries.

2) Since the year 2011 movies and series has increased rapidly in number but till 2018 after which a fall can be seen till the year 2021

3) The director based on this data who has directed more number of movies and series is Rajiv Chilakha

4) Most of the shows has got the ratings of TV-MA, TV-14 and R which tells us that the platform is not suitable for kids.